guw/main.py

```diff
@@ -187,27 +206,42 @@ def _sync_at(self, tmpdir, backup, local, features, prev_feature):
         last_feature = self.config["features"][-1]
         if last_feature:
             if last_feature["status"] != "integrated":
-                target_branch_name = self.config["target"]["branch"]
-                last_feature_branch = f"{last_feature['remote']}/{last_feature['name']}"
-                logger.info(f"Making target branch {target_branch_name} based on {last_feature_branch}")
-                repo.git.checkout("-b", target_branch_name, last_feature_branch)
-                if backup:
-                    feature_backup_name = self._backup_name(target_branch_name)
-                    logger.debug(f"Backing up target branch into {feature_backup_name}")
-                    repo.git.branch("-c", feature_backup_name)
-                    self.to_push.append((feature_backup_name, self.config["target"]["remote"]))
-                repo.git.reset("--hard", last_feature["name"])
-                self.to_push.append((target_branch_name, self.config["target"]["remote"]))
+                self._copy(repo, last_feature, self._get_target_feature(), backup)
             else:
                 logger.info("All features already integrated, nothing to do")
+        # If we are syncing from upstream, make sure to update source too
+        upstream_feature = self._get_upstream_feature()
+        if (
+            source_name == upstream_feature["name"]
+            and source_remote == upstream_feature["remote"]
+            and not self._upstream_is_source()
+        ):
+            source_feature = self._get_source_feature()
+            # Rename the upstream branch to avoid the case the source and upstream
+            # share the same branch name (origin/main and upstream/main)
+            upstream_feature_name = f"upstream-{upstream_feature['name']}"
+            repo.git.branch("-M", upstream_feature["name"], upstream_feature_name)
+            upstream_feature["name"] = upstream_feature_name
+            self._copy(repo, upstream_feature, source_feature, backup)
```

Comment:
can we add `repo.git.checkout(self.config["target"]["branch"])` here.

Sometimes I keep the temporal dir to check the process is ok. I would like to have it pointing to the target. Therefore only `cd /tmp/tmpu_t8179w && git log` is needed  to check it.

Reply:
Isn't the copy(a, b, backup) enough?

Reply:
Ok, so what you want is that once a sync process ends, you want the target branch to be the active branch on the checkout?

Reply:
exactly